homeassistant/components/humidifier/__init__.py

```diff
@@ -134,23 +140,23 @@ def state_attributes(self) -> dict[str, Any]:
     @property
     def target_humidity(self) -> int | None:
         """Return the humidity we try to reach."""
-        return None
+        return self._attr_target_humidity
 
     @property
     def mode(self) -> str | None:
         """Return the current mode, e.g., home, auto, baby.
 
         Requires SUPPORT_MODES.
         """
-        raise NotImplementedError
+        return self._attr_mode
```

Comment:
This one is mandatory. We should raise when it's not set. We could use a sentinel

```python
_SENTINEL = object()

class HumidifierEntity(ToggleEntity):
    _attr_mode = _SENTINEL

    @property
    def mode(self):
        if self._attr_mode is _SENTINEL:
            raise NotImplementedError

        return self._attr_mode
```

Reply:
Oh nevermind, you didn't set a default value so it will crash. oops.